### ISSUE TITLE
Very slow interface in a low end Windows computer.

### ISSUE THREAD

**Issue by mdio33** [airsonic-advanced/airsonic-advanced#441]
## Problem description

I have a small file server running windows server with a Atom D525, 6Gb Ram, SO and airsonic-advanced running in a SSD
I know it's a very limited hardware, but it woks for file server and it have low power consumption.
But airsonic interface is so slow that is almost unusable... with patience i can sellect the music and after it starts playing, it works very well, no stop or buffering.

Is there any way to optimize the system to get better performance?
What are the minimum requirements to run airsonic-advantage with good performance?

I'm the only user.

### Steps to reproduce

1. Navigating in web interface, Settigns...

## System information

 * **Airsonic version**: 11.0.0-SNAPSHOT.20210119054517
 * **Operating system**: Windows Server 2016
  * **Java version**: Java SE Deve kit 14.0.1 (64Bit)
 * **Proxy server**: None
 * **Client**: Firefox, Chorme
 * **Language**: English

## Additional notes

Thank you for the work.

**Comment by Trufax** [comment]
Did you check the resource consumption during the browsing of the webinterface? I

**Comment by mdio33** [comment]
> 
> 
> Did you check the resource consumption during the browsing of the webinterface? I
Yes, OpenJDK Platform Binary sits around 25%-30% CPU (2 cores, 4 threads) on login and until home appears (it takes at leaset 15s) then If i click "Settings" CPU rise again and it take at least 2 minutes (yes 120s) to display Settings page, then click in "Personal" it takes at least 1m30 to display.

Never go up than 30%
![imagem](https://user-images.githubusercontent.com/44021265/105104843-f073e780-5aaa-11eb-91a1-5ce34b9651db.png)

**Comment by Trufax** [comment]
Oh so you run the server on a windows PC`? Do you use the same PC to access the web interface ? Did you check if the indexing is running? What does the browser say (F12 debug console)?

**Comment by IwishIcanFLighT** [comment]
I installed a fresh instance of Airsonic Advanced yesterday on my server and can confirm I have the same issue loading the settings pages or the More page. The UI seems to hang a good second or two before loading these pages. The rest of the UI (exploring media, playlists, starred, about, etc) is loading just fine.

Looking at my CPU usage when reproducing the issue, I can clearly see a good 25% usage peak. Nothing in the debug console.

**Comment by mdio33** [comment]
> 
> 
> Oh so you run the server on a windows PC`? Do you use the same PC to access the web interface ? Did you check if the indexing is running? What does the browser say (F12 debug console)?

It's the same, I can use the server to access the web interface ou my personal computer, that is equal slow, in the example i was using my personal computer with Windows 10 to browse the interface, but the printscreen was from the server with Windows Server 2016.
No, the index is not running.
I can look to Debug Console, but I don't know what I should look for...

I been trying with other older versions of JDK but the its the same.

**Comment by Trufax** [comment]
Hm than iam sorry but i cannot help you. I run it under Linux. Maybe increasing the RAM allocation for the JVM helps.

**Comment by IwishIcanFLighT** [comment]
I also run it on Linux (Unraid via Docker). CPU is an i5 750 (old but still decent for this task) and I assigned 1GB of RAM.

**Comment by mdio33** [comment]
> 
> 
> Hm than iam sorry but i cannot help you. I run it under Linux. Maybe increasing the RAM allocation for the JVM helps.

I explained at the beginning that it was installed on a Windows Server ...

And I had assigned 1Gb too, with the "-Xmx1024m" option but the process stays in between 400-500 Mb

**Comment by Trufax** [comment]
You are right dont know how i could missed that. So you use Open JDK maybe try the Oracle one ? Did you try the normal airsonic (not advanced) does it run slow as well ?

**Comment by mdio33** [comment]
> 
> 
> Hm than iam sorry but i cannot help you. I run it under Linux. Maybe increasing the RAM allocation for the JVM helps.

I explained at the beginning that it was installed on a Windows Server ...

And I had assigned 1Gb too, with the "-Xmx1024m" option but the process stays in between 400-500 Mb

**Comment by mdio33** [comment]
> 
> 
> You are right dont know how i could missed that. So you use Open JDK maybe try the Oracle one ? Did you try the normal airsonic (not advanced) does it run slow as well ?
The original airsonic 10.6.2 is a little slow in the first time, nothing like airsonic-adv, sellecting settings take a 5-10 secs, but then everything runs smooth.

Going to try a fresh config.

**Comment by mdio33** [comment]
A clean config is equal slow... I guess my little server is no good for airsonic-adv, its strange because when I started using it, it was perfectly usable, for more than a year now.
Going to test with and older release or get back to the original airsonic.

**Comment by jobenvil** [comment]
> I guess my little server is no good for airsonic-adv

Don't think so, if it worked earlier, should work now better with the improvements. Somethings is going under the table or we miss it. If you could recognice the version which had the performance impact, will be very helpful - I guess.

**Comment by mdio33** [comment]
> 
> 
> > I guess my little server is no good for airsonic-adv
> 
> Don't think so, if it worked earlier, should work now better with the improvements. Somethings is going under the table or we miss it. If you could recognice the version which had the performance impact, will be very helpful - I guess.

I don't know why, but I tested a few releases to the last 10.0.6 and I equaly slow... maybe I'm confused with the airsonic (non-adv).

I been trying also many versions of JRE, the oldest one that work is this https://jdk.java.net/java-se-ri/11 

Can you advise what is the best version to use?

More, I only use Airsonic to stream MP3, is there any config I could do to optimize the speed?

Thank you once again.

**Comment by mdio33** [comment]
Just for the record...

Subsonic - Super fast, very snappy as it should in any option of the web interface, settings or media.
Airsonic - Slow in settings, fast in media navigation and playing.
Airsonic Advantage - Very very slow in settings,  fast in media navigation and playing.

**Comment by dsiminiuk** [comment]
Subsonic - no releases since November, 2019.
Airsonic - no releases since May, 2020.

Bug fixes and enhancement requests for both go unanswered and stale.

Airsonic-Advanced seems to be the only one advancing.

**Comment by mdio33** [comment]
> 
> 
> Subsonic - no releases since November, 2019.
> Airsonic - no releases since May, 2020.
> 
> Bug fixes and enhancement requests for both go unanswered and stale.
> 
> Airsonic-Advanced seems to be the only one advancing.

I understand that, but it's a big diference.

I'll stick with Subsonic for now, like I said, I only want to play MP3, but I will keep an eye in Airsonic Advanced, to check if it get better abour performance and if I decide to upgrade my little server. :)

Thank you all.

**Comment by jobenvil** [comment]
Maybe you want to compare [my network performance HAR-File from Chrome dev console](https://cloud.hiperborea.de/f/cd560bc4e70043d59198/?dl=1) with yours. You can import the file and maybe you can discover some network issue. I would suggest playing with the chrome dev features (network, performance, memory)

**Comment by mdio33** [comment]
> 
> 
> Maybe you want to compare [my network performance HAR-File from Chrome dev console](https://cloud.hiperborea.de/f/cd560bc4e70043d59198/?dl=1) with yours. You can import the file and maybe you can discover some network issue. I would suggest playing with the chrome dev features (network, performance, memory)

Can you give me a tip how to do that?? 😊

**Comment by dsiminiuk** [comment]
@mdio33 I play mp3 files all day long on Airsonic-Advanced. I see no difference between it and legacy versions.

**Comment by TheFlatworm** [comment]
Subsonic uses Jetty vs Airsonics Tomcat, big difference in speed on slower machines.  Test with subsonic standalone vs running in tomcat.

**Comment by mdio33** [comment]
> 
> 
> Subsonic uses Jetty vs Airsonics Tomcat, big difference in speed on slower machines. Test with subsonic standalone vs running in tomcat.

Now that makes sense. ty

**Comment by clach04** [comment]
Possibly a case of #602?

**Comment by mdio33** [comment]
Thank you for the answer, but i decided to upgrade my server, that is running now on a Xeon E5-2630L and everything is running smoth. :)
